style: apply /style-guide pass to models/sweeps (#2722)
wandb/docs: models/sweeps/existing-project.mdx

title: 'Tutorial: Create sweep job from project'
---

> This tutorial explains how to create sweep jobs from a pre-existing W&B project. By the end, you'll have created a baseline project, configured a hyperparameter sweep, and launched agents that run training jobs in parallel. You use the [Fashion MNIST dataset](https://github.com/zalandoresearch/fashion-mnist) to train a PyTorch convolutional neural network to classify images. The [W&B examples repository (PyTorch CNN Fashion)](https://github.com/wandb/examples/tree/master/examples/pytorch/pytorch-cnn-fashion) provides the required code and dataset.

Reviewer: ```suggestion
This tutorial explains how to create sweep jobs from a pre-existing W&B project. It guides you to create a baseline project, configure a hyperparameter sweep, and launch agents that run training jobs in parallel. 

You use the [Fashion MNIST dataset](https://github.com/zalandoresearch/fashion-mnist) to train a PyTorch convolutional neural network to classify images. The [W&B examples repository (PyTorch CNN Fashion)](https://github.com/wandb/examples/tree/master/examples/pytorch/pytorch-cnn-fashion) provides the required code and dataset.
```

Author: Applied your suggestion (split the intro into two paragraphs).

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

Explore the results in this [W&B Dashboard](https://app.wandb.ai/carey/pytorch-cnn-fashion).

## Create a project

First, create a baseline project by training the example model at least once. This baseline gives the sweep something to configure against in later steps. Download the PyTorch MNIST dataset example model from the W&B examples GitHub repository. Next, train the model. The training script is within the `examples/pytorch/pytorch-cnn-fashion` directory.

> To download and train the example model, follow these steps:

Reviewer: ```suggestion
To download and train the example model:
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

1. Clone the repository: `git clone https://github.com/wandb/examples.git`.
2. Open the example directory: `cd examples/pytorch/pytorch-cnn-fashion`.

> 3. Run the training script manually: `python train.py`.

Reviewer: These would be better as code blocks than inline

Author: Converted the inline commands in each step to `bash` code blocks.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

> Optional: Explore the example in the W&B App dashboard. [View an example project page](https://app.wandb.ai/carey/pytorch-cnn-fashion).

Reviewer: This would be another step

Author: Made the "Optional: Explore..." line step 4 of the procedure.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

After this initial run completes, you have a baseline project in W&B that the sweep can build on.

## Create a sweep

With a baseline project in place, you can configure a sweep over its runs. From your project page, open the [**Sweep** tab](/models/sweeps/visualize-sweep-results) in the project sidebar and select **Create Sweep**.

<Frame>
    <img src="/images/sweeps/sweep1.png" alt="W&B project page with the Sweep tab open and the Create Sweep button highlighted" />
</Frame>

The auto-generated configuration suggests values to sweep over based on the runs you've completed. Edit the configuration to specify what ranges of hyperparameters you want to try. When you launch the sweep, it starts a new process on W&B's hosted sweep server. This centralized service coordinates the agents (the machines that run the training jobs).

<Frame>
    <img src="/images/sweeps/sweep2.png" alt="Auto-generated sweep configuration editor showing hyperparameter ranges" />
</Frame>

## Launch agents

> After you configure the sweep, launch one or more agents locally to execute the runs. To distribute the work and finish the sweep job more quickly, launch up to 20 agents on different machines in parallel. The agent prints out the next set of parameters to use.

Reviewer: This could be streamlined, the first two sentences feel like they are saying a similar thing.

Author: Streamlined so the sentences no longer overlap: launch one or more agents, then launch up to 20 in parallel across machines to finish faster, then each agent prints the next set of parameters.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

<Frame>
    <img src="/images/sweeps/sweep3.png" alt="Terminal output from a sweep agent printing the next set of hyperparameters" />
</Frame>

> You now have a running sweep that coordinates training jobs across your agents and reports results back to W&B. The following image shows what the dashboard looks like as the example sweep job runs.

Reviewer: ```suggestion
You now have a running sweep that coordinates training jobs across your agents and reports results back to W&B. The following image show the dashboard with the example sweep job running.
```

Author: Applied your suggestion (fixed "show" to "shows").

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

<Frame>
    <img src="/images/sweeps/sweep4.png" alt="Sweep dashboard plotting metrics across parallel training runs" />
</Frame>

## Seed a new sweep with existing runs

> You can also launch a new sweep using existing runs that you've previously logged, which lets you reuse earlier results as a starting point. To seed a new sweep with existing runs, follow these steps:

Reviewer: ```suggestion
To reuse earlier results as a starting point, launch a new sweep using existing runs that you've previously logged:
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

1. Open your project table.

> 2. Select the runs you want to use by enabling their row checkboxes.

Reviewer: ```suggestion
2. Click a run row's checkbox to select the run.
```

Author: Applied your suggestion.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

3. Select the dropdown to create a new sweep.

Your sweep is now set up on the server. Launch one or more agents to start the runs.

<Frame>
    <img src="/images/sweeps/tutorial_sweep_runs.png" alt="Project runs table with rows selected and the create sweep option in the dropdown" />
</Frame>

<Note>

> If you start the new sweep as a Bayesian sweep, the selected runs also seed the Gaussian Process.

Reviewer: Why is Process capitalized? Not something I'm super familiar with.

Author: Lowercased to "Gaussian process" — "Gaussian" is the proper adjective and "process" is a common noun.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
